metabase/metabase
# Migrate snippet form to formik

Epic #26178

1. Migrates `SnippetModal` to `formik`
2. Breaks `SnippetModal` into `SnippetForm` and `SnippetModalForm`
3. Covers both modal and form with unit-tests
4. Removes form-related code from the snippets entity

## To Verify

> **Note**
> Form behavior can be different depending on if you're running an OSS or paid version of Metabase. Paid Metabase plans allow to create snippet folders (called snippet collections internally) to organize snippets. So for OSS, you shouldn't see a folder picker at all. In paid plans, you should see one **if you've created at least one folder**

### OSS

1. New > SQL question
2. Click the "snippets" icon at the right side of the query content editor (right above the "Run" button)
3. Click "Create snippet"
4. Ensure you can't submit the form until both name and content are filled in
5. Ensure you can save and use the snippet
6. Once you've created at least one snippet, click on it in the snippets sidebar and then click "Edit"
7. Ensure you can rename and change content/description for a snippet
8. Ensure you can archive the snippet using the "Archive" button at the bottom left of the snippet editing modal (also make sure you can't see this button when creating a new snippet)

### Paid

1. Spin up paid Metabase and sign in as admin
2. New > SQL question
3. Click the "snippets" icon at the right side of the query content editor (right above the "Run" button)
4. Click the "+" icon at the top right of the snippets sidebar and pick "new folder"
5. Create a folder
6. Test out both snippet creating and editing flows and ensure you can pick a snippet folder now

## Demo

https://user-images.githubusercontent.com/17258145/205351179-992ec560-e9dd-473a-9acf-af4ff95f6eab.mp4
